docs/content/guides/custom-http-headers.md

```diff
@@ -14,11 +14,162 @@ None
+
+```js
+// app.json
+{
+ expo: {
+  //...
+ },
+ plugins: [
+  ['./plugins/withPlugin.ts', {
+   headers: [
+    ["authorization", "long-hash-string"],
+    ["Some-other-header", "value"],
+   ],
+  }],
+  // ...
+ ],
+ // ...
+}
+```
+
+```ts
+// ./plugins/withPlugin.ts
+// Inspired by https://github.com/invertase/react-native-firebase/blob/main/packages/app/plugin/src/ios/appDelegate.ts
+
+import configPlugin, { type ConfigPlugin } from '@expo/config-plugins'
+import { type AppDelegateProjectFile } from '@expo/config-plugins/build/ios/Paths'
+import { mergeContents } from '@expo/config-plugins/build/utils/generateCode'
+import fs from 'fs'
+
+const { IOSConfig, WarningAggregator, withDangerousMod } = configPlugin
+
+function modifySwiftAppDelegate(contents: string, headers: Header[]): string {
+    let methodInvocationBlock = `\t\tMLRNCustomHeaders().initHeaders()`
+    headers.forEach((value) => {
+        methodInvocationBlock += `\n\t\tMLRNCustomHeaders().addHeader("${value[1].replaceAll('"', '\\"')}", forHeaderName: "${value[0]}")`
+    })
+    // MLRNCustomHeaders().addHeader(
+    const methodInvocationLineMatcher =
+        /(?:self\.moduleName\s*=\s*"([^"]*)")|(?:factory\.startReactNative\()/
+
+    // Add import
+    if (!contents.includes('import maplibre_react_native')) {
+        contents = contents.replace(
+            /import Expo/g,
+            `import Expo
+import maplibre_react_native`,
+        )
+    }
+    if (!methodInvocationLineMatcher.test(contents)) {
+        WarningAggregator.addWarningIOS(
+            'custom headers plugin',
+            'Unable to determine correct insertion point in AppDelegate.swift. Skipping addition.',
+        )
+        return contents
+    }
+
+    // Add invocation
+    return mergeContents({
+        tag: 'custom header plugin',
+        src: contents,
+        newSrc: methodInvocationBlock,
+        anchor: methodInvocationLineMatcher,
+        offset: 0, // new line will be inserted right above matched anchor
+        comment: '//',
+    }).contents
+}
+
+async function modifyAppDelegateAsync(
+    appDelegateFileInfo: AppDelegateProjectFile,
+    headers: Header[],
+) {
+    const { contents, path } = appDelegateFileInfo
+    let newContents = modifySwiftAppDelegate(contents, headers)
+    await fs.promises.writeFile(path, newContents)
+}
+
+const withAppDelegate: ConfigPlugin<Props> = (config, { headers }) => {
+    return withDangerousMod(config, [
+        'ios',
+        async (config) => {
+            const fileInfo = IOSConfig.Paths.getAppDelegate(
+                config.modRequest.projectRoot,
+            )
+            await modifyAppDelegateAsync(fileInfo, headers)
+            return config
+        },
+    ])
+}
+
+type Header = [string, string]
+
+type Props = {
+    headers: Header[]
+}
+
+const withPlugin: ConfigPlugin<Props> = (config, props) =>
+    withAppDelegate(config, props)
+
+export default withPlugin
+```
+
+#### Swift `AppDelegate.mm`
```

Comment:
Just for clarification, this is the manual variant, basically doing the same like Expo Plugin? And it's basically adding the `import` and the two `MLRNCustomHeaders()` lines to the default Expo template right? I think this might be better understandable when formatted as a `diff`.

Reply:
Correct, I changed it to diff. I also added warning regarding potential leaking of authorization tokens into another layers.